diff/parse.go

```diff
@@ -136,9 +136,75 @@ func parseFileHeader(line string) (filename, timestamp string) {
 	ss := line[len(tokenOldFile)+1:]
 	tabi := strings.LastIndex(ss, "\t")
 	if tabi == -1 {
-		return ss, ""
+		return unescapeCString(ss), ""
 	}
-	return ss[:tabi], ss[tabi+1:]
+	return unescapeCString(ss[:tabi]), ss[tabi+1:]
+}
+
+func unescapeCString(str string) string {
```

Comment:
Do you have any references or something to implement this function?

Reply:
It from quote.c of https://github.com/git/git
https://github.com/git/git/blob/77556354bb7ac50450e3b28999e3576969869068/quote.c#L345-L413

git-apply and git-diff use quote.c

Reply:
Interesting. Thank you for the information.

Can you add the reference as a comment and optionally left some context coment?